### DISCUSSION TITLE
p-fileUpload not working properly after updrage angular, primeng version of project from 14 to 17

### DISCUSSION THREAD

**Discussion by puzsarbalazs** [primefaces/community#1426] [PrimeNG]
After the updrage to version 17, p-fileUpload not wokring porperly, if its mode is advanced, and auto=true. The file uploading is not beginning automatically after the file selection, and because of the auto=true, there is no upload button, so the file cant be uploaded in any way.

`<p-fileUpload [auto]="true" [mode]="'advanced'"></p-fileUpload>`

The other 3 combinations are working properly (basic-auto=true, basic-auto=false, advanced-basic=false)

**Comment by cetincakiroglu** [comment]
Hi,

Could you please share a stackblitz example so we can identify the issue?

**Reply by puzsarbalazs** [reply]
Hi, yes, in this stackblitz example there are all four combinations and it shows that 3 of them is working, and the advanced auto=true is not.
https://stackblitz.com/edit/ql9kiq?file=src%2Fapp%2Fdemo%2Ffile-upload-auto-demo.html

**Reply by wessisaac** [reply]
I am also facing the same issue, after upgraded to PrimeNg 17, this issue is occurred. Expecting solution from PrimeNg team.

**Reply by wessisaac** [reply]
The awkward temporary solution is, 

1. In the html's p-fileupload, set mode="advanced" initially.
    <p-fileUpload mode="advanced" [auto]="true" (onSelect)="onSelect($event)" (onUpload)="onUpload($event)" #fileInput url="{{url}}">
    <ng-template let-file pTemplate="content">
     .
     .
     </ng-template>
     </p-fileUpload>

2.  In the .ts file, at the end of onSelect method, please change the mode as "basic" temporarily, to fire the "onUpload" event
     this.fileInput.mode="basic";
3. In the .ts file, at the end of onUpload($event) method, again set the initial mode "advanced" to utilize advanced feature.
    this.fileInput.mode="advanced";

**Reply by puzsarbalazs** [reply]
My temporary solution was that on the end of onSelect method if the mode is advanced and auto=true then I call the onUpload method
